solr/core/src/java/org/apache/solr/util/tracing/SimplePropagator.java

```diff
@@ -0,0 +1,304 @@
+      return this;
+    }
+
+    @Override
+    public SpanBuilder asChildOf(SpanContext parent) {
+      this.parent = parent;
+      return this;
+    }
+
+    @Override
+    public SpanBuilder asChildOf(Span parent) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder ignoreActiveSpan() {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder withStartTimestamp(long arg0) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String arg0, String arg1) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String arg0, boolean arg1) {
+      return this;
+    }
+
+    @Override
+    public SpanBuilder withTag(String arg0, Number arg1) {
+      return this;
+    }
+
+    @Override
+    public <T> SpanBuilder withTag(Tag<T> arg0, T arg1) {
+      return this;
+    }
+  }
+
+  private static final class SimplePropagatorSpanContext implements SpanContext {
+
+    private final String traceId;
+
+    public SimplePropagatorSpanContext(String traceId) {
+      this.traceId = traceId;
+    }
+
+    @Override
+    public String toTraceId() {
+      return traceId;
+    }
+
+    @Override
+    public String toSpanId() {
+      return "";
+    }
+
+    @Override
+    public Iterable<Map.Entry<String, String>> baggageItems() {
+      return Collections.emptyList();
+    }
+
+    @Override
+    public String toString() {
+      return "SimplePropagatorSpanContext [traceId=" + traceId + "]";
+    }
+  }
+
+  static final class SimplePropagatorSpan implements Span {
+
+    private final SpanContext ctx;
+
+    private SimplePropagatorSpan(SpanContext ctx) {
+      this.ctx = ctx;
+    }
+
+    @Override
+    public SpanContext context() {
+      return ctx;
+    }
+
+    @Override
+    public void finish() {}
+
+    @Override
+    public void finish(long arg0) {}
+
+    @Override
+    public String getBaggageItem(String arg0) {
+      return null;
+    }
+
+    @Override
+    public Span log(Map<String, ?> arg0) {
+      return this;
+    }
+
+    @Override
+    public Span log(String arg0) {
+      return this;
+    }
+
+    @Override
+    public Span log(long arg0, Map<String, ?> arg1) {
+      return this;
+    }
+
+    @Override
+    public Span log(long arg0, String arg1) {
+      return this;
+    }
+
+    @Override
+    public Span setBaggageItem(String arg0, String arg1) {
+      return this;
+    }
+
+    @Override
+    public Span setOperationName(String arg0) {
+      return this;
+    }
+
+    @Override
+    public Span setTag(String arg0, String arg1) {
+      return this;
+    }
+
+    @Override
+    public Span setTag(String arg0, boolean arg1) {
+      return this;
+    }
+
+    @Override
+    public Span setTag(String arg0, Number arg1) {
+      return this;
+    }
+
+    @Override
+    public <T> Span setTag(Tag<T> arg0, T arg1) {
+      return this;
+    }
+  }
+
+  private static String newTraceId() {
```

Comment:
nit: this method seems to hide down here after these long inner subclasses.  Above them is probably better organized.